Algorithm.Framework/Portfolio/AccumulativeInsightPortfolioConstructionModel.cs

```diff
@@ -0,0 +1,243 @@
+        {
+            _percent = Math.Abs(percent);
+        }
+
+       
+        /// <summary>
+        /// Method that will determine if the portfolio construction model should create a
+        /// target for this insight
+        /// </summary>
+        /// <param name="insight">The insight to create a target for</param>
+        /// <returns>True if the portfolio should create a target for the insight</returns>
+        public virtual bool ShouldCreateTargetForInsight(Insight insight)
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Will determine the target percent for each insight
+        /// </summary>
+        /// <param name="activeInsights">The active insights to generate a target for</param>
+        /// <returns>A target percent for each insight</returns>
+        public Dictionary<Insight, double> DetermineTargetPercent(ICollection<Insight> activeInsights)
+        {
+            var result = new Dictionary<Insight, double>();
+
+            foreach (var insight in activeInsights)
+            {
+                if (usedInsight.ContainsKey(insight))
+                {
+                   continue;
+                }
+                usedInsight[insight] = 1;
+                if (positionSizes.ContainsKey(insight.Symbol))
+                {
+                   positionSizes[insight.Symbol] += _percent * (int)insight.Direction;
+                }
+                else
+                {
+                     positionSizes[insight.Symbol] = _percent * (int)insight.Direction;
+                }
+
+                if (insight.Direction == 0)
+                {
+                   // We received a Flat
+
+                   // if adding or subtracting will push past 0, then make it 0
+                   if (Math.Abs(positionSizes[insight.Symbol]) < _percent)
+                   {
+                      positionSizes[insight.Symbol] = 0;
+                   }
+
+                   // otherwise, we flatten by percent
+                   if (positionSizes[insight.Symbol] > 0)
+                   {
+                        positionSizes[insight.Symbol] -= _percent;
+                   }
+                   if (positionSizes[insight.Symbol] < 0)
+                   {
+                      positionSizes[insight.Symbol] += _percent;
+                   }
+                }
+            
+                result[insight] = positionSizes[insight.Symbol];
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Will determine the target percent for each expired insight (since expired insights
+        /// will remove the amount accumulated in portfolio)
+        /// </summary>
+        /// <param name ="activeInsights">The insights that are expiring and must be updated
+        /// <returns>A target percent for each insight</returns>
+        public Dictionary<Insight, double> UpdateExpiredInsights(ICollection<Insight> activeInsights)
```

Comment:
Does this need to be public? Also, `activeInsights` can be an `IEnumerable<Insight>` instead.